storage/remote/queue_manager.go

```diff
@@ -802,12 +917,12 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan sample) {
 
 		case sample, ok := <-queue:
 			if !ok {
-				if nPending > 0 {
-					level.Debug(s.qm.logger).Log("msg", "Flushing samples to remote storage...", "count", nPending)
-					s.sendSamples(ctx, pendingSamples[:nPending], &buf)
-					s.qm.metrics.pendingSamples.Sub(float64(nPending))
-					level.Debug(s.qm.logger).Log("msg", "Done flushing.")
+				if nPending == 0 {
+					s.queueBuf[shardID] <- nil
```

Comment:
Should this close the channel instead of writing a nil marker? Same with down below, close the channel once done writing to it.

Reply:
Yes, that is better.